profile-universal/src/checks/transformed_components.rs

```diff
@@ -87,3 +97,135 @@ fn transformed_components(f: &Testable, context: &Context) -> CheckFnResult {
+    new_font.add_table(&new_loca).map_err(|x| x.to_string())?;
+    new_font.copy_missing_tables(f.font());
+    let new_bytes = new_font.build();
+    std::fs::write(&t.filename, new_bytes).map_err(|_| "Couldn't write file".to_string())?;
+
+    Ok(true)
+}
+
+fn decompose_glyph(
+    composite: &CompositeGlyph,
+    glyphs: &[WriteGlyph],
+) -> Result<WriteGlyph, String> {
+    let mut new_glyph = SimpleGlyph::default();
+    for component in composite.components() {
+        for (gid, affine) in flatten_component(glyphs, component)? {
+            let component_glyph = glyphs.get(gid.to_u16() as usize).ok_or("glyph not found")?;
+            match component_glyph {
+                WriteGlyph::Simple(simple) => {
+                    new_glyph
+                        .contours
+                        .extend(simple.contours.iter().map(|c| transform_contour(c, affine)));
+                }
+                _ => {
+                    panic!("unexpected glyph type")
+                }
+            }
+        }
+    }
+    Ok(WriteGlyph::Simple(new_glyph))
+}
+
+fn transform_contour(c: &Contour, affine: Affine) -> Contour {
+    c.iter()
+        .map(|point| {
+            let kurbo_pt = kurbo::Point::new(point.x as f64, point.y as f64);
+            let new_pt = affine * kurbo_pt;
+            CurvePoint::new(new_pt.x as i16, new_pt.y as i16, point.on_curve)
+        })
+        .collect::<Vec<_>>()
+        .into()
+}
+
+fn flatten_component(
+    glyphs: &[WriteGlyph],
+    component: &Component,
+) -> Result<Vec<(GlyphId16, kurbo::Affine)>, String> {
+    let glyph = glyphs
+        .get(component.glyph.to_u16() as usize)
+        .ok_or("glyph not found")?;
+    Ok(match glyph {
+        WriteGlyph::Empty => vec![],
+        WriteGlyph::Simple(_) => vec![(
+            component.glyph,
+            to_kurbo_transform(&component.transform, &component.anchor),
+        )],
+        WriteGlyph::Composite(composite_glyph) => {
+            let mut all_flattened_components = vec![];
+            for component in composite_glyph.components() {
+                all_flattened_components.extend(flatten_component(glyphs, component)?.iter().map(
+                    |(gid, transform)| {
+                        let new_transform =
+                            to_kurbo_transform(&component.transform, &component.anchor)
+                                * *transform;
+                        (*gid, new_transform)
+                    },
+                ));
+            }
+            all_flattened_components
+        }
+    })
+}
+
+fn to_kurbo_transform(transform: &Transform, anchor: &Anchor) -> kurbo::Affine {
+    let (dx, dy) = match anchor {
+        Anchor::Offset { x, y } => (*x, *y),
+        Anchor::Point { .. } => (0, 0),
+    };
+    kurbo::Affine::new([
+        transform.xx.to_f32() as f64,
+        transform.xy.to_f32() as f64, // check
+        transform.yx.to_f32() as f64, // check
```

Comment:
What did you mean by "check" in these two comments? Does it mean you were unsure about the correctness of those 2 lines?

Reply:
Yes. Kurbo's affine transform is in a matrix like so:

```
| a c e |
| b d f |
| 0 0 1 |
```

I *think* `b` here is a scale_xy, but I'm not totally sure.

I also realized that this implementation doesn't hotfix `gvar`, which is a problem.